api/app/db/crud/weather_models.py

```diff
@@ -252,6 +252,30 @@ def get_station_model_predictions(
     return query
 
 
+def get_station_model_prediction_from_previous_model_run(
+        session: Session,
+        station_codes: List[int],
+        model: str,
+        prediction_timestamp: str,
+        prediction_model_run_timestamp: str):
+    """ Fetches the one model prediction for the specified station_code, model, and prediction_timestamp
+    from the prediction model run immediately previous to the given prediction_model_run_timestamp.
+    """
+    # create a lower_bound for time range so that we're not querying timestamps all the way back to the
+    # beginning of time
+    lower_bound = str(prediction_model_run_timestamp - datetime.timedelta(days=1))
```

Comment:
Shouldn't be a string - this code should be fine.
```suggestion
    lower_bound =prediction_model_run_timestamp - datetime.timedelta(days=1)
```